src/routes/common/interceptors/route-logger.interceptor.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  NestInterceptor,
+} from '@nestjs/common';
+import { ILoggingService, LoggingService } from '../logging/logging.interface';
+import { Inject } from '@nestjs/common/decorators';
+
+/**
+ * The {@link RouteLoggerInterceptor} is an interceptor that logs the requests
+ * that target a specific route.
+ *
+ * Since this is an interceptor we have access to the respective {@link ExecutionContext}
+ * See https://docs.nestjs.com/fundamentals/execution-context
+ */
+@Injectable()
+export class RouteLoggerInterceptor implements NestInterceptor {
+  constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+  ) {}
+
+  intercept(context: ExecutionContext, next: CallHandler) {
+    const req = context.switchToHttp().getRequest();
+    // TODO Use req.route.path to log the route path without any replacement
```

Comment:
> Just curious about this `TODO`. Is this something you want to change in the future? Do you see any difference between using `req.url` and `req.route.path`?

We don't need to replace it but at least to add it since it contains the route path without any parameters replaced. This is useful for logging and gathering metrics for each route (without doing custom parsing on log ingestion).

I've added the comment since I didn't want to change the format of the logs in this PR.

Reply:
Great, makes sense, thanks for the explanation 🙂

Reply:
There's an example which shows what `req.route.path` represents at the end of the PR description 😉